src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -275,6 +275,21 @@ protected override ImmutableArray<PendingBranch> Scan(ref bool badRegion)
             Analyze(compilation, method, node, diagnostics, useMethodSignatureReturnType: false, useMethodSignatureParameterTypes: false, methodSignatureOpt: null, returnTypes: null, initialState: null, callbackOpt);
         }
 
+        internal static void AnalyzeIfNeeded(
+            CSharpCompilation compilation,
+            BoundAttribute attribute,
+            DiagnosticBag diagnostics)
+        {
+            if (compilation.LanguageVersion < MessageID.IDS_FeatureNullableReferenceTypes.RequiredVersion() ||
+                attribute.Constructor is null ||
+                !compilation.SyntaxTrees.Contains(attribute.SyntaxTree)) // don't analyze speculative attributes
```

Comment:
>!compilation.SyntaxTrees.Contains(attribute.SyntaxTree) [](start = 16, length = 55)

This is not a sufficient check for a speculative SemanticModel. Instead consider what I suggested in Binder_Attributes.cs